### ISSUE TITLE
Don't assume http address is 127.0.0.1:8500

### ISSUE THREAD

**Issue by ghostsquad** [WeAreFarmGeek/diplomat#157]
Configuration should look for environment variable `CONSUL_HTTP_ADDR` for address/port, unless explicitly configured.

https://medium.com/zendesk-engineering/making-docker-and-consul-get-along-5fceda1d52b9

> How about Consul API access? The key is to set two standard environment variables called CONSUL_HTTP_ADDR and CONSUL_RPC_ADDR. Nearly all standard Consul client libraries use these values to determine where to send queries. Be certain your code uses these variables too — never hard-code Consul endpoints into your applications!
```
$ sudo docker run --dns 169.254.1.1 \
              -e CONSUL_HTTP_ADDR=169.254.1.1:8500 \
              -e CONSUL_RPC_ADDR=169.254.1.1:8400 ...
```

**Comment by taharah** [comment]
I'm not opposed to this change; however, it would most certainly be a pretty big breaking change. I'll take a look at some of the other libraries available for Consul, and if it is adopted elsewhere, then I'll go ahead and get this staged up for a 3.x release in the future.

**Comment by ghostsquad** [comment]
Not all of them do this from my research, however, this is the functionality of the native Consul CLI.
https://www.consul.io/api/libraries-and-sdks.html
https://www.consul.io/docs/commands/index.html

Like the AWS CLI, it's nice when client libraries follow the same configuration behavior as their native CLI counterparts, as it means consumers don't need to re-configure the client, and can rely on already established environment variables or config files (e.g. `AWS_ACCESS_KEY_ID` and `~/.aws/credentials`)

**Comment by taharah** [comment]
So I recently saw how they were setting the default values for the Vault ruby gem [[1](https://github.com/hashicorp/vault-ruby/blob/master/lib/vault/defaults.rb)] and I would like to implement the same for Diplomat. I've started looking at how we can implement this (hopefully without any breaking changes).

**Comment by pierresouchay** [comment]
Fully agree on the feature, might break existing usages however :(
Since we merged #163, maybe for instances with configuration, setting this value by default would make sense?
